stubs/setuptools/pkg_resources/__init__.pyi

```diff
@@ -359,7 +360,7 @@ def evaluate_marker(text: str, extra: Incomplete | None = None) -> bool: ...
 class NullProvider:
     egg_name: str | None
     egg_info: str | None
-    loader: types._LoaderProtocol | None
+    loader: LoaderProtocol | None
```

Comment:
@abravalheri Sorry I just noticed. But you won't be able to make 3rd-party stubs changes in the same PR. Until the next mypy & pyright release, as they'll need to first include the new stdlib changes from typeshed.

pyright is released quite often. So realistically this just means waiting for mypy for a follow-up PR. I can open a PR to update `stubs/setuptools/pkg_resources/__init__.pyi` as soon as this one is merged. And keep it on hold until the next version of mypy.

Reply:
Wait, wasn't this part the suggestion in https://github.com/python/typeshed/pull/11890#discussion_r1596813098?

If `types._LoaderProtocol` has to be maintained, than it makes almost no difference to replace `types._LoaderProtocol` because it is only used in 2 places: internally in `types` and in `pkg_resources`... And then we should probably just simply revert https://github.com/python/typeshed/pull/11890/commits/7075e5e9607a8bd98912a7f9ebb3bf65cbfa1314 and https://github.com/python/typeshed/pull/11890/commits/45b7a5c9ae08beeefd06fb80a1238a3ca1174446.

Reply:
>  But you won't be able to use new typeshed symbols in 3rd-party stubs in the same PR

Is this documented in the CONTRIBUTING guide? Sorry I might have missed that part.

Reply:
You can replace all usages of `types._LoaderProtocol` in stdlib in this PR. Just not for 3rd-party stubs. Otherwise uses of the most recent `types-setuptools` will have a reference to `_typeshed.importlib.LoaderProtocol` in their stub that doesn't exist yet.

`types._LoaderProtocol` should still exist until mypy updates their vendored typeshed stdlib stubs (which is done every version).

When I made that comment, I didn't think about the usage in setuptools stubs. Since it's *only* used there and in `types`, it might be cleaner to just revert as you're suggesting. And to remove `types_LoaderProtocol` along the follow-up PR that will use `_typeshed.importlib.LoaderProtocol` in setuptools stubs.

Sorry for the "flip-flopping" here ^^"

> Is this documented in the CONTRIBUTING guide? Sorry I might have missed that part.

I don't think so, but it probably should, now that you mention it. It's rare, but we've been bitten by it twice (that I can remember) in the past.

Reply:
It doesn't look like this comment from Avasam was addressed before the PR was merged — I think our setuptools stubs are now importing a protocol from `_typeshed` which, from mypy's perspective, doesn't exist yet. I think we need to change this so that the `LoaderProtocol` definition is temporarily duplicated in our `setuptools` stubs, rather than being imported from `_typeshed`, until a version of mypy is released that includes this protocol in its vendored stdlib stubs

Reply:
https://github.com/python/typeshed/pull/11909